copy Rt episode in middle of delays episodes (#37)
epiverse-trace/tutorials-early: episodes/quantify-transmissibility.Rmd

---
title: 'Quantifying transmission'
teaching: 30
exercises: 0
---

```{r setup, echo = FALSE, warning = FALSE, message = FALSE}
library(EpiNow2)
library(ggplot2)
withr::local_options(list(mc.cores = 4))
```

:::::::::::::::::::::::::::::::::::::: questions 

- How can I estimate the time-varying reproduction number ($Rt$) and growth rate from a time series of case data?

> - How can I quantify geographical heterogeneity in these transmission metrics? 

Author: ```suggestion
- How can I quantify geographical heterogeneity from these transmission metrics? 
```

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

- Learn how to estimate transmission metrics from a time series of case data using the R package `EpiNow2`

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: prereq

## Prerequisites

> Learners should familiarise themselves with following concept dependencies before working through this tutorial: 

Author: ```suggestion
Learners should familiarise themselves with following concepts before working through this tutorial: 
```

**Statistics** : probability distributions, principle of Bayesian analysis. 

> 

Author: added in #41

> 

Author: solved in https://github.com/epiverse-trace/tutorials-early/pull/37/commits/beba32c9311e102b64285c4344e0b3dd44288804

> **Epidemic theory** : Effective reproduction number.

Author: added in #41

> **Epidemic theory** : Effective reproduction number.

Author: solved in https://github.com/epiverse-trace/tutorials-early/pull/37/commits/beba32c9311e102b64285c4344e0b3dd44288804

:::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: callout
### Reminder: the Effective Reproduction Number, $R_t$ 

The [basic reproduction number](../learners/reference.md#basic), $R_0$, is the average number of cases caused by one infectious individual in a entirely susceptible population. 

But in an ongoing outbreak, the population does not remain entirely susceptible as those that recover from infection are typically immune. Moreover, there can be changes in behaviour or other factors that affect transmission. When we are interested in monitoring changes in transmission we are therefore more interested in the value of the **effective reproduction number**, $R_t$, the average number of cases caused by one infectious individual in the population at time $t$.

::::::::::::::::::::::::::::::::::::::::::::::::

## Introduction

> Quantifying transmission metrics at the start of an outbreak can give important information on the strength of transmission (reproduction number) and the speed of transmission ([growth rate](../learners/reference.md#growth), doubling/halving time). To estimate these key metrics using case data we must account for delays between the date of infections and date of reported cases. In an outbreak situation, data are usually available on reported dates only, therefore we must use estimation methods to account for these delays when trying to understand changes in transmission over time. 

Author: ```suggestion
The transmission intensity of an outbreak is quantified using two key metrics: the reproduction number, which informs on the strength of the transmission by indicating how many new cases are expected from each existing case; and the [growth rate](../learners/reference.md#growth), which informs on the speed of the transmission by indicating how rapidly the outbreak is spreading or declining (doubling/halving time) within a population. To estimate these key metrics using case data we must account for delays between the date of infections and date of reported cases. In an outbreak situation, data are usually available on reported dates only, therefore we must use estimation methods to account for these delays when trying to understand changes in transmission over time. 
```

> In the next tutorials we will focus on how to implement the functions in `{EpiNow2}` to estimate transmission metrics of case data. We will not cover the theoretical background of the models or inference framework, for details on these concepts see the [vignette](https://epiforecasts.io/EpiNow2/dev/articles/estimate_infections.html).

Author: ```suggestion
In the next tutorials we will focus on how to use the functions in `{EpiNow2}` to estimate transmission metrics of case data. We will not cover the theoretical background of the models or inference framework, for details on these concepts see the [vignette](https://epiforecasts.io/EpiNow2/dev/articles/estimate_infections.html).
```

> For more details on the distinction between speed and strength of transmission and implications for control, see [Dushoff & Park, 2021](https://royalsocietypublishing.org/doi/full/10.1098/rspb.2020.1556).

Author: ```suggestion

```

::::::::::::::::::::::::::::::::::::: callout
### Bayesian inference

The R package `EpiNow2` uses a [Bayesian inference](../learners/reference.md#bayesian) framework to estimate reproduction numbers and infection times based on reporting dates.

In Bayesian inference, we use prior knowledge (prior distributions) with data (in a likelihood function) to find the posterior probability.

<p class="text-center" style="background-color: white">Posterior probability $\propto$ likelihood $\times$ prior probability
</p>

::::::::::::::::::::::::::::::::::::::::::::::::

> :::::::::::::::::::::::::::::::::::::::::::::::: instructor

Author: no, but we can try to add titles to them to link or redirect to them. Is that the feature you request?

> We refer to the Prior probability distribution and the [Posterior probability](https://en.wikipedia.org/wiki/Posterior_probability) distribution.

Author: ```suggestion
We refer to the prior probability distribution and the [posterior probability](https://en.wikipedia.org/wiki/Posterior_probability) distribution.
```

Lines below, in the "`Expected change in daily cases`" callout, by "the posterior probability that $R_t < 1$", we refer specifically to the [area under the posterior probability distribution curve](https://www.nature.com/articles/nmeth.3368/figures/1). 

::::::::::::::::::::::::::::::::::::::::::::::::

The first step is to load the `{EpiNow2}` package :

```{r, eval = FALSE}
library(EpiNow2)
```

:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::: instructor

This tutorial illustrates the usage of `epinow()` to estimate the time-varying reproduction number and infection times. Learners should understand the necessary inputs to the model and the limitations of the model output. 

::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

## Delay distributions and case data 
### Case data

To illustrate the functions of `EpiNow2` we will use outbreak data of the start of the COVID-19 pandemic from the United Kingdom. The data are available in the R package `{incidence2}`. 

```{r}
dplyr::as_tibble(incidence2::covidregionaldataUK)
```

To use the data, we must format the data to have two columns:

> + `date` : the date (as a date object see `?is.Date()`),

Author: solved in https://github.com/epiverse-trace/tutorials-early/pull/37/commits/beba32c9311e102b64285c4344e0b3dd44288804

> + `confirm` : number of confirmed cases on that date.

Author: added in #41

> + `confirm` : number of confirmed cases on that date.

Author: solved in https://github.com/epiverse-trace/tutorials-early/pull/37/commits/beba32c9311e102b64285c4344e0b3dd44288804

Let's use `{tidyr}` and `incidence2::incidence()` for this:

> ```{r, warning = FALSE, message = FALSE}

Author: agree for consistency. At least for packages out of the main ones that belong to the main objective of the episode.

> ```{r, warning = FALSE, message = FALSE}

Author: Added in #42

> ```{r, warning = FALSE, message = FALSE}

Author: solved in https://github.com/epiverse-trace/tutorials-early/pull/37/commits/beba32c9311e102b64285c4344e0b3dd44288804

ggplot(cases[1:90, ], aes(x = date, y = confirm)) +
  geom_col() +
  theme_grey(
    base_size = 15
  )
```

### Delay distributions 
We assume there are delays from the time of infection until the time a case is reported. We specify these delays as distributions to account for the uncertainty in individual level differences. The delay can consist of multiple types of delays/processes. A typical delay from time of infection to case reporting may consist of :

<p class="text-center" style="background-color: aliceblue">**time from infection to symptom onset** (the [incubation period](../learners/reference.md#incubation)) + **time from symptom onset to case notification** (the reporting time)
.</p>

The delay distribution for each of these processes can either estimated from data or obtained from the literature. We can express uncertainty about what the correct parameters of the distributions by assuming the distributions have **fixed** parameters or whether they have **variable** parameters. To understand the difference between **fixed** and **variable** distributions, let's consider the incubation period. 

::::::::::::::::::::::::::::::::::::: callout
### Delays and data

> The number of delays and type of delay is a flexible input that depends on the data. The examples below highlight how the delays can be specified for different data sources:

Author: ```suggestion
The number of delays and type of delay are a flexible input that depend on the data. The examples below highlight how the delays can be specified for different data sources:
```

<center>

| Data source        | Delay(s) |
| ------------- |-------------|
|Time of symptom onset      |Incubation period |
|Time of case report      |Incubation period + time from symptom onset to case notification |
|Time of hospitalisation   |Incubation period + time from symptom onset to hospitalisation     |

</center>

::::::::::::::::::::::::::::::::::::::::::::::::

#### Incubation period distribution 

> The distribution of incubation period can usually be obtained from the literature. The package `{epiparameter}` contains a library of epidemiological parameters for different diseases obtained from the literature. 

Author: ```suggestion
The distribution of incubation period for many diseases can usually be obtained from the literature. The package `{epiparameter}` contains a library of epidemiological parameters for different diseases obtained from the literature. 
```

+ The exponential growth rate of case numbers is `r summary(estimates)$estimate[summary(estimates)$measure=="Rate of growth"]`.

+ The doubling time (the time taken for case numbers to double) is `r summary(estimates)$estimate[summary(estimates)$measure=="Doubling/halving time (days)"]`.

::::::::::::::::::::::::::::::::::::: callout
### `Expected change in daily cases` 

A factor describing expected change in daily cases based on the posterior probability that $R_t < 1$.

<center>
| Probability ($p$)      | Expected change |
| ------------- |-------------|
|$p < 0.05$    |Increasing |
|$0.05 \leq p< 0.4$    |Likely increasing |
|$0.4 \leq p< 0.6$    |Stable |
|$0.6 \leq p < 0.95$    |Likely decreasing |
|$0.95 \leq p \leq 1$    |Decreasing |
</center>

::::::::::::::::::::::::::::::::::::::::::::::::

## Quantify geographical heterogeneity

The outbreak data of the start of the COVID-19 pandemic from the United Kingdom from the R package `{incidence2}` includes the region in which the cases were recorded. To find regional estimates of the effective reproduction number and cases, we must format the data to have three columns:

+ `date` : the date,

> + `region` : the region, 

Reviewer: Remove the space before the colons ":"

Author: added in #41

+ `confirm` : number of confirmed cases for a region on a given date.

```{r}
regional_cases <-
  incidence2::covidregionaldataUK[, c("date", "cases_new", "region")]
colnames(regional_cases) <- c("date", "confirm", "region")

# extract the first 90 dates for all regions
dates <- sort(unique(regional_cases$date))[1:90]
regional_cases <- regional_cases[which(regional_cases$date %in% dates), ]

head(regional_cases)
```

To find regional estimates, we use the same inputs as `epinow()` to the function `regional_epinow()`:

```{r, message = FALSE, eval = TRUE}
estimates_regional <- regional_epinow(
  reported_cases = regional_cases,
  generation_time = generation_time_opts(generation_time_fixed),
  delays = delay_opts(
    incubation_period_fixed + reporting_delay_fixed
  ),
  rt = rt_opts(
    prior = list(mean = rt_log_mean, sd = rt_log_sd)
  )
)

estimates_regional$summary$summarised_results$table

estimates_regional$summary$plots$R
```

## Summary

> `EpiNow2` can be used to estimate transmission metrics from case data at the start of an outbreak. The reliability of these estimates depends on the quality of the data and appropriate choice of delay distributions. In the next tutorial we will learn how to make forecasts and investigate some of the additional inference options available in `EpiNow2`. 

Reviewer: Why you keep emphasizing the point `at the start of an outbreak`? I sought this package can be used any time in course of the outbreak period. Note that It calculated the effective (not the basic) reproduction number.

Author: ```suggestion
`EpiNow2` can be used to estimate transmission metrics from case data at any time in the course of an outbreak. The reliability of these estimates depends on the quality of the data and appropriate choice of delay distributions. In the next tutorial we will learn how to make forecasts and investigate some of the additional inference options available in `EpiNow2`. 
```
